### PR TITLE
Players can choose their name

### LINKED ISSUE #527: [FEATURE] Players can choose their name
## 🚀 Feature

A player should be able to choose a name for themselves (maybe in main lobbies screen?) and then we will propagate that to the games so they show up instead of the player IDs (not replace, just for UI stuff, internally should be IDs)

### PR DESCRIPTION
**Pull request by ncontinanza**
Closes #527 

A player can now choose their name when starting the game, and it will be kept until it's changed.

Things to test:

1. Start the game and make sure a pop up appears, asking you to enter your name.
2. Start a new game, and observe you're no longer called "Player 1", but the name you just picked. This applies to the lobby, the character selection scene, the match and the death splash.
3. Close the application and open it again. This time you shouldn't get the pop up, as your name was saved in the device you're using.
4. Use the button in the lobbies scene to change your name and verify it's working correctly.
5. Play a multiplayer match (remember to do it in your localhost) and check that the name assignments are being made correctly.

### CODE REVIEW

**Review by BertovDev — CHANGES_REQUESTED** [review]
I tested the functionality its nice! and i have some comments that i believe they will improve much more this:

- Player name should be shown somewhere in the screen. So you know which one you have selected.
- Input placeholder should show the current name you have: example: If i change the name to `Berto` and then i wanted to change it again the placeholder shouldnt say Player Name
Captura de pantalla 2023-10-25 a la(s) 16 36 25

- Name length should be restricted to a certain amount of characters. If not i can type infinite and the UI will be broken.

- When I enter a game and then disconnect and reconnect an error is been displayed and the name reset to player {id}
Captura de pantalla 2023-10-25 a la(s) 16 30 58



I have notice this issues just playing around with it, I didnt check any code yet.
What do you think about what I commented?

**Comment by saugon** [comment]
There are some fix proposals that have arisen from playtesting:

- [ ] Add padding space between the text input and the error message.
Enhance players' names in battle (the ones that appear above the players' characters):
  - [ ] Increase the size of the name box to prevent two-line names.
  - [ ] Set a maximum character name size.
  - [ ] Reduce the font size.
- [ ] The placeholder shouldn't be included in the input value (when tapping the input, it's annoying to have to erase it).

**Review by BertovDev — COMMENTED** [review]
Looks awesome to me and i will only add a method called `GetPlayerNameById` but only if this functionality will persevere in the future (since we will use another way of storing the names)

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Captura de pantalla 2023-10-30 a las 11 56 13 a  m

The input text should be uppercase.

File: client/Assets/Scenes/Lobbies.unity
[the remark this answers is not in the document]
Comment (ncontinanza) [reply client/Assets/Scenes/Lobbies.unity:1]: Done!

**Comment by ncontinanza** [comment]
> The input text should be uppercase.

That is the user's input, right? Currently the user is able to use lowercase, and we change it to uppercase afterwards. If you submit a name that contains lowercase letters, you should have it converted to uppercase everywhere, including that placeholder the next time you click on the change name button.

**Review by BertovDev — DISMISSED** [review]
Love this PR!

**Comment by AngieDutra** [comment]
> > The input text should be uppercase.
> 
> That is the user's input, right? Currently the user is able to use lowercase, and we change it to uppercase afterwards. If you submit a name that contains lowercase letters, you should have it converted to uppercase everywhere, including that placeholder the next time you click on the change name button.

I commented on it from a design POV. We are using that font only in uppercase format. If uppercase is not intended in this input I would recommend changing the font.

**Comment by saugon** [comment]
> > > The input text should be uppercase.
> > 
> > 
> > That is the user's input, right? Currently the user is able to use lowercase, and we change it to uppercase afterwards. If you submit a name that contains lowercase letters, you should have it converted to uppercase everywhere, including that placeholder the next time you click on the change name button.
> 
> I commented on it from a design POV. We are using that font only in uppercase format. If uppercase is not intended in this input I would recommend changing the font.

This is a valid observation. I wouldn't change the font but would display the text always in uppercase characters.
In any case, I'd set this aside for now and address it in a future PR if needed.

**Review by BertovDev — APPROVED** [review]

**Review by AngieDutra — APPROVED** [review]
This PR is awesome ✨ LGTM

**Pull request merged**